Dockerfile

```diff
@@ -207,6 +229,86 @@ RUN mkdir src &&\
     cmake \
         --build src/build \
         --target install \
+#	-j$(nproc) \
     &&\
     ln -s ${__prefix}/bin/thisdd4hep.sh ${__ldmx_env_script_d__}/thisdd4hep.sh &&\
     rm -r src
+
+###############################################################################
+# LHAPDF
+#
+# Needed for GENIE
+#
+###############################################################################
+LABEL lhapdf.version="6.5.2"
+ENV PYTHON_VERSION=3
+RUN mkdir src &&\
+    ${__wget} https://lhapdf.hepforge.org/downloads/?f=LHAPDF-6.5.2.tar.gz |\
+     ${__untar} &&\
+    cd src &&\
+    ./configure --prefix=${__prefix} &&\
+    make -j$(nproc) &&\
+    make -j$(nproc) install &&\
+    cd ../ &&\
+    rm -rf src
+
+###############################################################################
+# Installing log4cpp
+#
+# Needed for GENIE
+#
+###############################################################################
+
+RUN apt-get update &&\
+    apt-get install -y autoconf automake libtool
+
+RUN mkdir src &&\
+    ${__wget} https://sourceforge.net/projects/log4cpp/files/latest/download |\
+    ${__untar} &&\
+    cd src/log4cpp &&\
+    ./autogen.sh &&\
+    ./configure &&\
+    make check &&\
+    make install -j$(nproc) &&\
+    cd ../.. &&\
+    rm -r src
+
+###############################################################################
+# GENIE
+#
+# Needed for ... GENIE :)
+#
+###############################################################################
+LABEL genie.version=3.02.00
+ENV GENIE_VERSION=3_02_00
+ENV GENIE_REWEIGHT_VERSION=1_02_00
+
+ENV GENIE=/usr/local/GENIE/Generator
+
+RUN mkdir -p /usr/local/root &&\
+    ln -s /usr/local/lib/root /usr/local/root/lib &&\
+    export ROOTSYS=/usr/local/root &&\
+    mkdir -p /usr/local/GENIE &&\
+    cd /usr/local/GENIE &&\
+    git clone --branch R-$GENIE_VERSION https://github.com/GENIE-MC/Generator.git &&\
+    cd Generator &&\
+    export LD_LIBRARY_PATH=/usr/local/lib:/usr/local/lib/root &&\
+    ./configure --enable-lhapdf6 --disable-lhapdf5 \
+                --enable-gfortran --with-gfortran-lib=/usr/x86_64-linux-gnu/ \
+                --disable-pythia8 --with-pythia6-lib=${__prefix}/pythia6 \
+                --enable-test && \
+    make -j$(nproc)
```

Comment:
Ah... I see why you need `git`. Does GENIE not have an "install" option? i.e. Do we need to carry around all of its code even after it is built?

That might be the case and if so, I would use the following `wget` command. The URL for the tar-ball was manually looked up on the GENIE-MC GitHub.

```
wget -q -O - https://github.com/GENIE-MC/Generator/archive/refs/tags/R-${GENIE_VERSION}.tar.gz | tar -xz
```

Reply:
There are some install options for GENIE, but without doing a lot of digging I think it will be difficult to get everything all consistent, particularly installation of configuration/tune information. I've gone ahead and also done a make install here, but think that we should keep the GENIE area around and available in the container to support runtime needs, and to keep things structured such that GENIE internal programs that may be making bad assumptions at least know where to look for things still.

I've made the change from git to wget. It's unfortunately likely that we will at some point need to use an unofficial / not fully distributed version of GENIE to support various activities, but I think that we can cross that bridge when we get there. Perhaps in that case we make/keep tags on

Reply:
We've had to move to a custom fork of Geant4 for LDMX as well. I would suggest we adopt the same approach with GENIE, where we maintain a fork of it with our customizations. Then we can make our own releases/tags that the Dockerfile can download from.